File: src/main/java/com/springexample/irrigationagriculture/api/AdminApi.java
package com.springexample.irrigationagriculture.api;

import com.springexample.irrigationagriculture.entity.enums.TimeZone;
import com.springexample.irrigationagriculture.exception.GeneralException;
import com.springexample.irrigationagriculture.exception.exceptions.NoUserFoundException;
import com.springexample.irrigationagriculture.service.AdminService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("admin")
public class AdminApi {

    private final AdminService adminService;

    public AdminApi(AdminService adminService) {
        this.adminService = adminService;
    }


    @PostMapping("/changeCommand")
    public ResponseEntity<String> changeCommand(@RequestHeader("Authorization") String token, @RequestParam String username, @RequestParam Boolean value)
            throws GeneralException, NoUserFoundException {

        return ResponseEntity.ok(adminService.changeCommand(token,username,value));

    }

    @PostMapping("/assignScheduler")
    public ResponseEntity<String> assignScheduler(@RequestHeader("Authorization")
     String token,@RequestParam String username,@RequestParam String time,@RequestParam String time2) throws GeneralException, NoUserFoundException {

        return ResponseEntity.ok(adminService.assignScheduler(token,username,time,time2));

    }

    @PostMapping("/addUser")
    public ResponseEntity<String> addUser(@RequestHeader("Authorization") String token,@RequestParam String username) throws GeneralException, NoUserFoundException {

        return ResponseEntity.ok(adminService.addUser(token,username));

    }


    @GetMapping("/enterRoom")
    public Boolean enterAdminRoom(){

       return adminService.enterAdminRoom();

    }


}

File: app/src/main/java/com/example/irrigationfrontend/api/AdminApi.kt
package com.example.irrigationfrontend.api

import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Query

data class AddUserRequest(
    val username: String
)

data class ChangeCommandRequest(
    val username: String,
    val grant: Boolean
)

interface AdminApi {
    @GET("admin/enterRoom")
    fun enterRoom(@Header("Authorization") token: String): Call<ResponseBody>

    @POST("admin/addUser")
    fun addUser(
        @Header("Authorization") token: String,
        @Query("username") username: String
    ): Call<ResponseBody>

    @POST("admin/changeCommand")
    fun changeCommand(
        @Header("Authorization") token: String,
        @Query("username") username: String,
        @Query("value") value: Boolean
    ): Call<ResponseBody>

    @POST("admin/assignScheduler")
    fun assignScheduler(
        @Header("Authorization") token: String,
        @Query("username") username: String,
        @Query("time") time: String,
        @Query("time2") time2: String
    ): Call<ResponseBody>
}
